File: Backend/src/main/java/hr/bornast/fantasy/domain/model/Match.java
package hr.bornast.fantasy.domain.model;

import java.time.OffsetDateTime;
import java.util.HashSet;
import java.util.Set;

import lombok.Data;

@Data
public class Match {
    private int id;
    private MatchTeam homeTeam;
    private MatchTeam awayTeam;
    private Set<MatchGoal> goals = new HashSet<>();
    private Set<MatchCard> cards = new HashSet<>();
    private League league;
    private OffsetDateTime matchDate;
    private Referee referee;
    private Stadium stadium;
    private int spectatorCount;

    public void addGoal(MatchGoal goal) {
        goals.add(goal);
    }

    public void addCard(MatchCard card) {
        cards.add(card);
    }
}

File: Frontend/src/app/models/match.ts
import { RecordName } from "./recordName";

export class Match {
    id: number;
    homeTeam: MatchTeam;
    awayTeam: MatchTeam;
    goals: MatchGoal[];
    cards: MatchCard[];
    league: RecordName;
    matchDate: Date;
    referee: RecordName;
    stadium: RecordName;
    spectatorCount: number;
    name: string;
    result: string;
}

export class MatchTeam {
    team: RecordName;
    formation: RecordName;
    lineupPlayers: MatchPlayer[];
    substitutePlayers: MatchPlayer[];
    substitutions: MatchSubstitution[];
    coach: RecordName;
}

export class MatchSubstitution {
    lineupPlayer: MatchPlayer;
    substitutePlayer: MatchPlayer;
    minute: number;
}

export class MatchPlayer {
    id: number;
    name: string;
    rate: number;
}

export class MatchGoal {
    player: RecordName;
    minute: number;
}

export class MatchCard {
    player: RecordName;
    card: RecordName;
    minute: number;
}
